### PR TITLE
add variables fixes

### PR DESCRIPTION
**Pull request by jakekaplan**
Ensure that if under the `variables` schema of `Workpool.base_job_template`, if there is a required field that has a default, the required-ness of the field is ignored because we can expect the field to be populated via the default

### CODE REVIEW

File: tests/server/schemas/test_actions.py [hunk 1]
```diff
@@ -78,6 +79,48 @@ def test_create_with_worker_pool_name_warns(self, kwargs):
...
+                    "my-field": {
+                        "type": "string",
+                        "title": "My Field",
+                    },
+                },
+            }
+        }
+        with pytest.raises(jsonschema.ValidationError) as excinfo:
+            deployment_create.check_valid_configuration(base_job_template)
+        assert excinfo.value.message == "'my-field' is a required property"
```
[the remark this answers is not in the document]
Comment (jakekaplan) [reply tests/server/schemas/test_actions.py:103]: can't hurt! let me add

[the remark this answers is not in the document]
Comment (serinamarie) [reply tests/server/schemas/test_actions.py:103]: thanks! it would give me peace of mind anyway 😅

[the remark this answers is not in the document]
Comment (serinamarie) [reply tests/server/schemas/test_actions.py:103]: oh, I actually meant one that didn't have a default value but was required

[the remark this answers is not in the document]
Comment (serinamarie) [reply tests/server/schemas/test_actions.py:103]: but maybe that's not likely

**Review by serinamarie — APPROVED** [review]

**Pull request merged**